plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerSchema.java

```diff
@@ -535,11 +535,13 @@ protected SQLServerTableIndex fetchObject(JDBCSession session, SQLServerSchema o
                     indexType = DBSIndexType.OTHER;
                     break;
             }
+            boolean isColumnstoreIndex = indexTypeNum == 5 || indexTypeNum == 6;
```

Comment:
Please don't add magic values. Let's define a constants for them.
I see that we have already got a switch with such magic values, but let's stop it.

Reply:
I added it here. But I'm not sure that we need to move all documentations stuff in our code :(

Reply:
What for?

Reply:
Introducing an enum would be overkill since these values are only used in a single place.

Reply:
It's hard to support. I should always return back to the comment and read it again to know what's the meaning for each number.
Besides that, I don't understand what is this code for when I read `indexTypeNum  == 5`, it's hard to understand.

Reply:
These values have been already used in two places.

Reply:
okay, at least just a constants...

Reply:
We can create SQL Server extra index types, but in this case, my condition will be not
`index.isColumnStore` 
but 
`index.getIndexType == SQLServerIndexType.COLUMNSTOREClustered || index.getIndexType == SQLServerIndexType.COLUMNSTORENonClustered`

And these extra indexes need to be also like clustered/nonclustered children... I think it is overcomplicated.

indexTypeNum == 5 - Now in the comment. But I still think from my side that documentation is always our friend. And this is not the best idea to keep this sometimes outdated info in DBeaver.

Reply:
Perhaps she meant introducing private constants:

```java
private static final int INDEX_TYPE_HEAP = 0; 
private static final int INDEX_TYPE_CLUSTERED_ROWSTORE = 1;
private static final int INDEX_TYPE_NONCLUSTERED_ROWSTORE = 2;
private static final int INDEX_TYPE_XML = 3;
private static final int INDEX_TYPE_SPATIAL = 4;
private static final int INDEX_TYPE_CLUSTERED_COLUMNSTORE = 5;
private static final int INDEX_TYPE_NONCLUSTERED_COLUMNSTORE = 6;
private static final int INDEX_TYPE_NONCLUSTERED_HASH = 7;
```

It's not overkill since these would be inlined after compilation.

Reply:
Yes, I meant what @ShadelessFox proposed.

Reply:
Thank you.